JahsieAult/homework.md

```diff
@@ -0,0 +1,164 @@
+
+### Homework #1
+(Due Monday 1/12)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+>True. The value of the string and the value of the number are equivalent. 
+
+**b)** `"1" === 1`
+>False. The values are the same but they are not the same AND of the same type. 
+
+**c)** `x == 'x'`
+>False. The first is a variable and the second is a string of 'x'. The variable could hold any value. This would be true if the value of x was "x"
+
+**d)** `x == (x+'')`
+>True. Adding an empty string concatenates nothing to the variable. This leaves it unchanged. 
+
+**e)** `'' == ' '`
+>False. The first value is empty but the second contains a space. 
+
+**f)** `x = true`
+>True. We are assigning a boolean true. So it returns the truth. 
+
+**g)** `var x; x == 'undefined'`
+>False. 'var x' does not assign a value, so x retains whatever unknown value it previously had. "x == 'undefined'" is a string. So this will only be true if x = 'undefined' (not the same as x = undefined).
+
+**h)** `'9'<'10'`
+>False. Because these are strings it evaluates the first character and then quits. Since 9 > 1 it decides that the boolean is false.
+
+**i)** `typeof x + 1 === "number"`
+>False. Without parentheses this will return number1 and interpret these as two separate elements. The parentheses are required for the console to interpret them as a unit. 
+
+**j)** `typeof x % 2 === "number"`
+>False. For the same reason as above.
+
+**k)** `typeof (x % 2) === "number"`
```

Comment:
1k) Exactly! And just for clarity, if `x` is not a number, `(x % 2) == NaN`, which (strangely) is a number.